### DISCUSSION TITLE
Update v0.4.0 - Vue.js 3.0 & Typescript 4.1 & MapboxGL 2.0

### DISCUSSION THREAD

**Discussion by alkaitagi** [alkaitagi/Aevzag#85] [General]
### New storage
The most notable change is that now all JSON is being stored and subsequently served from the storage branch on the repo. I believe this separation of linguistic data from the main app source allows for a cleaner local workflow and easier content pull requesting. Also, all content modifications (by committing or pull requesting into the storage branch) will now appear on the site in under 5 minutes as per GitHub's caching timeout, without the need for the app recompilation.

### Local environment
When in production mode, the app fetches files directly from the branch, and when in development mode, it fetches them from the local path, same a before. To enable local testing, copy the storage branch into your local copy of the main branch, under the static folder. It will not leak into the main branch due to the .gitignore update, as the main branch now contains only the app source.

### UI improvements

- Reworked map, so it now zooms smoothly and uses vector tiles, which results in sharped image and improved performance.
- Unified and simplified navigation and global controls across all the editor headers.
- Converter Editor now has a small text field where you can test your mappings right away.
- Added the ability to reorder lists across various parts of the editors.
- Removed unnecessary Phrasebook display controls, so notes and interactivity are always enabled, as well as cleaned up context display.

### Breaking JSON changes

- In **catalogue.json** property `coordinates` was renamed into `point` and the numbers are swapped.
- In **converter.json** property `default` was deleted. Now the default conversion pair is always the 0th and the 1st mappings.
- In **phrasebook.json** (corpus) phrase's `preview` was renamed into `name`.

**Comment by alkaitagi** [comment]
In particular, I want to hear feedback regarding that new store branch and language contents separation.

**Reply by parryc** [reply]
I think it's a good idea. For posterity, I was able to set it up locally with `git worktree add -b store public/store upstream/store`. Not to mention, it appears nicely separated on the GitHub UI.

**Reply by alkaitagi** [reply]
I've just released a small update that makes the storage addressing a bit more convenient. Now the app will always address the web store (located at the `store` branch) unless there's a local `.env.local` file (in the root of the project) with the `VUE_APP_STORE=local` line in it.

Because over the course of the latest developments I've realized that actual change of JSON format is not quite frequent, and most of the time the work doesn't involve experimenting with the local store. So it is just easier to always load the latest data from the remote branch and only explicitly specify the local store when there are some new linguistic functions.

**Comment by alkaitagi** [comment]
I've figured out, it would be better if I continue the remaining work on the phrasebook in the next v0.5.0 release, since that would introduce certain breaking changes in the JSON. Plus, I am not sure how much effort would it take to add audio playback capabilities to the phrasebook.

So, the next smaller release would focus on #83 and #86, after which I will actively work on the Phrasebook Corpus document.

**Reply by alkaitagi** [reply]
Oh, this is going to be more than just a slight update to the phrasebook's data structure. This will be a big overhaul paired with the editor's improved UX when it comes to context (and conditions) management.
